x/ibc-rate-limit/ibc_middleware_test.go

```diff
@@ -542,3 +543,22 @@ func (suite *MiddlewareTestSuite) TestUnsetRateLimitingContract() {
 	// N.B.: this panics if validation fails.
 	paramSpace.SetParamSet(suite.chainA.GetContext(), &params)
 }
+
+// Test rate limits are reverted if a "send" fails
+func (suite *MiddlewareTestSuite) TestNonICS20() {
+	suite.initializeEscrow()
+	// Setup contract
+	suite.chainA.StoreContractCode(&suite.Suite, "./bytecode/rate_limiter.wasm")
+	quotas := suite.BuildChannelQuota("weekly", "channel-0", sdk.DefaultBondDenom, 604800, 1, 1)
+	addr := suite.chainA.InstantiateRLContract(&suite.Suite, quotas)
+	suite.chainA.RegisterRateLimitingContract(addr)
+
+	osmosisApp := suite.chainA.GetOsmosisApp()
+
+	data := []byte("{}")
+	_, err := osmosisApp.RateLimitingICS4Wrapper.SendPacket(suite.chainA.GetContext(), capabilitytypes.NewCapability(1), "wasm.osmo1873ls0d60tg7hk00976teq9ywhzv45u3hk2urw8t3eau9eusa4eqtun9xn", "channel-0", clienttypes.NewHeight(0, 0), 1, data)
+
+	suite.Require().Error(err)
+	// This will error out, but not because of rate limiting
+	suite.Require().NotContains(err.Error(), "rate limit")
```

Comment:
Instead of notContains, could we just match the error that we get from this? Just in case we update the rate limit error in the future. 

Is a nit though.

Reply:
I was thinking the same but from a different direction: if the underlying error changes in ibc, this will continue to pass. But I can add the other check when I untangle the uncommitted changes on my current branch